consul/resolver.go

```diff
@@ -175,8 +306,16 @@ func (r *Resolver) discover() error {
 	}
 	r.mu.RUnlock()
 
+	apiStartTime := time.Now()
 	services, meta, err := r.client.Health().Service(serviceName, "", r.healthyOnly, queryOpts)
+	apiDuration := time.Since(apiStartTime)
+
+	r.recordAPIRequest(ctx, apiTypeHealthService, err == nil)
+	r.recordAPIDuration(ctx, apiTypeHealthService, apiDuration)
+
 	if err != nil {
+		r.recordError(ctx, serviceName, errorTypeConsulAPI)
```

Comment:
We do not need this as this info can be derived from `r.recordAPIRequest(ctx, apiTypeHealthService, err == nil)`